### ISSUE TITLE
relation_cloud_rs_zone_attach or zone not mapping subnet to AZ in Cloud ACI for AWS

### ISSUE THREAD

**Issue by marinalf** [CiscoDevNet/terraform-provider-aci#133]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.13.4
provider registry.terraform.io/ciscodevnet/aci v0.4.1

### APIC version and APIC Platform

5.0.2h on AWS 

### Affected Resource(s)

data "aci_cloud_cidr_pool"
resource "aci_cloud_cidr_pool"
resource "aci_cloud_subnet"

### Terraform Configuration Files

https://github.com/marinalf/cloudaci-demo-terraform/blob/main/main.tf

### Issue

Upon apply, when using only "relation_cloud_rs_zone_attach": 
 
```
resource "aci_cloud_subnet" "cloud_subnet" {
  cloud_cidr_pool_dn            = aci_cloud_cidr_pool.cloud_cidr_pool.id
  ip                            = "172.11.1.0/24"
  zone                          = "us-east-1a"
  relation_cloud_rs_zone_attach = "uni/clouddomp/provp-aws/region-us-east-1/zone-us-east-1a"
```

*Error: Invalid Configuration : All subnets under CtxProfile uni/tn-Prod/ctxprofile-ctx-vrf1-useast1 must either use zone property or CloudRsZoneAttach to specify the availability zone. Both under a single CtxProfile are not allowed*

When removing relation and adding "zone" instead, no more error, but subnet to mapping is still not happening. 

```
resource "aci_cloud_subnet" "cloud_subnet" {
  cloud_cidr_pool_dn            = aci_cloud_cidr_pool.cloud_cidr_pool.id
  ip                            = "172.11.1.0/24"
  zone                          = "us-east-1a"
```

### Expected Behavior

Subnet mapped to AZ under Context Profile

### Actual Behavior

Only subnet gets created in cAPIC

### Steps to Reproduce

Using ` relation_cloud_rs_zone_attach`or `zone` or both at the same time. It happens either using resource or data source for CIDR pool. 

### References

https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/cloudsubnet#zone

zone - (Optional) [AWS Only] Availability zone where the subnet must be deployed. This property can carry both the actual zone or the ACI logical zone name. In the former case, driver directly uses the value of this property. In the latter case, Connector has to first resolve the mapping from ACI logical zone to actual AWS zone. This parameter is required in APIC v5.0 or higher.

**Comment by nkatarmal-crest** [comment]
This issue has been fixed in the version 0.5.0 of the provider. Please try that and let me know if we can close this issue or not.

**Comment by marinalf** [comment]
Yes, the mapping is working now for both user and gateway subnets. The only thing missing is the name association per subnet. The name_alias gives the following error:

_Error: property nameAlias of uni/tn-Prod/ctxprofile-ctx-vrf1-useast1/cidr-[172.11.0.0/16]/subnet-[172.11.1.0/24] failed validation for value 'TGW subnet'_

```
resource "aci_cloud_subnet" "cloud_subnet1" {
  cloud_cidr_pool_dn            = data.aci_cloud_cidr_pool.cloud_cidr_pool.id
  ip                            = "172.11.1.0/24"
  usage                         = "gateway"
  zone                          = "uni/clouddomp/provp-aws/region-us-east-1/zone-us-east-1a"
  name_alias                    = "TGW subnet"
```

**Comment by nkatarmal-crest** [comment]
@marinalf name-alias is not supported this way for the subnet. @lhercot can help you with your use-case. Feel free to open new issue for the change you are looking for. I am closing out this one as the main issue mentioned in the description has been fixed in the v0.5.0
